tests/models/gpt2/test_modeling_gpt2.py

```diff
@@ -859,3 +868,40 @@ def test_contrastive_search_gpt2(self):
                 "but said in a statement to The Associated Press that"
             ],
         )
+
+    @require_flash_attn
+    @require_torch_gpu
+    @pytest.mark.flash_attn_test
+    @slow
+    def test_flash_attn_2_generate_padding_left(self):
```

Comment:
Doesn't the test in GPT2 already cover Decision Transformer? Since, basically the usage of Flash Attention in Decision Transformer happens exactly due to GPT2Model being embedded in its architecture

Reply:
Both model's should be tested. This makes sure that if anything changes upstream they remain correct, for example, inputs preparation in `DecisionTransformerModel`

Reply:
While adding the test for `DecisionTransformer` I realized that the model has two distinct `xxxPreTrainedModel`s and that adding support for `flash_attention_2` would be a bit more complicated, therefore I believe it would be better to have a specific PR to add support.

Reply:
In this case, flash attention shouldn't be added at all for the model. You can use `#Ignore copy` on the init so the previous attention class' method is used

Reply:
@amyeroberts https://github.com/huggingface/transformers/pull/29226/commits/94c2fe8859786fb1360584be352e06e9b4ff8ade